Sources/SwiftParser/Declarations.swift

```diff
@@ -1694,11 +1694,65 @@ extension Parser {
   ///     postfix-operator-declaration → 'postfix' 'operator' operator
   ///     infix-operator-declaration → 'infix' 'operator' operator infix-operator-group?
   ///     infix-operator-group → ':' precedence-group-name
+
+  struct OperatorDeclIntroducer {
+    var unexpectedBeforeFixity: RawUnexpectedNodesSyntax?
+    var fixity: RawTokenSyntax
+    var unexpectedBeforeOperatorKeyword: RawUnexpectedNodesSyntax?
+    var operatorKeyword: RawTokenSyntax
+  }
+
+  mutating func parseOperatorDeclIntroducer(_ attrs: DeclAttributes, _ handle: RecoveryConsumptionHandle) -> OperatorDeclIntroducer {
+    func isFixity(_ modifier: RawDeclModifierSyntax) -> Bool {
+      switch modifier.name {
+      case .keyword(.prefix),
+        .keyword(.infix),
+        .keyword(.postfix):
+        return true
+      default:
+        return false
+      }
+    }
+
+    var unexpectedBeforeFixity = RawUnexpectedNodesSyntax(attrs.attributes?.elements ?? [], arena: self.arena)
+
+    var fixity: RawTokenSyntax?
+    var unexpectedAfterFixity: RawUnexpectedNodesSyntax?
+
+    if let modifiers = attrs.modifiers?.elements {
+      if let firstFixityIndex = modifiers.firstIndex(where: { isFixity($0) }) {
+        let fixityModifier = modifiers[firstFixityIndex]
+        fixity = fixityModifier.name
+
+        let beforeFixity = Array(modifiers[0..<firstFixityIndex])
+
+        unexpectedBeforeFixity = RawUnexpectedNodesSyntax(combining: unexpectedBeforeFixity, RawUnexpectedNodesSyntax(beforeFixity, arena: self.arena), fixityModifier.unexpectedBeforeName, arena: self.arena)
+
+        unexpectedAfterFixity = RawUnexpectedNodesSyntax(combining: RawUnexpectedNodesSyntax(combining: fixityModifier.unexpectedBetweenNameAndDetail, RawUnexpectedNodesSyntax([fixityModifier.detail], arena: self.arena), fixityModifier.unexpectedAfterDetail, arena: self.arena), RawUnexpectedNodesSyntax(Array(modifiers[modifiers.index(after: firstFixityIndex)...]), arena: self.arena), arena: self.arena)
```

Comment:
To make this more readable, what do you think about adding a new overload of `RawUnexpectedNodesSyntax` that takes four parameters? Also, I think putting the parameters on separate lines would make this a lot more readable

```suggestion
        unexpectedAfterFixity = RawUnexpectedNodesSyntax(
          combining: fixityModifier.unexpectedBetweenNameAndDetail,
          RawUnexpectedNodesSyntax([fixityModifier.detail], arena: self.arena),
          fixityModifier.unexpectedAfterDetail,
          RawUnexpectedNodesSyntax(Array(modifiers[modifiers.index(after: firstFixityIndex)...]), arena: self.arena),
          arena: self.arena
        )
```